api/xml/elife/articles/00569.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" ?>
+<?xml-model href="../article.rng"?>
+
+<article xmlns="http://libero.pub" xmlns:elife="http://elifesciences.org" xml:lang="en">
+
+    <id>00569</id>
+    <doi>10.7554/eLife.00569</doi>
+    <title>The genome sequence of the colonial chordate, <elife:genome>Botryllus
```

Comment:
This `elife:` prefix and corresponding namespace seems to be the standard way for in-document extensions.

Reply:
It's not clear yet how this document can be produced. Several cases:
- standard mandatory fields like `<title>`: the article store can easily store them.
- standard optional fiels: same.
- extensions to the standard fields/new fields: the article store can expand what it accepts, but it needs to know about the full article schema. So if I want a IIIF service, it must get its schema extension into the article store somehow.
- other responses like `search`: sort them out separately, but they need to know the full schema too.

In a brute force way this would mean adding IIIF would mean redeploying the article service and potentially any dependants like `search`, creating a mutual dependency or a cascade of dependencies.

The possible options:
- having the article store use a standard article schema and have a `mashup` service add extensions from other services. This is unfeasible as it's too difficult to interpolate `<elife:genome>` in the right place.
- promote the concept of schema as data that is persisted rather than configuration. The article store should keep it somewhere in some kind of database, and use it to validate incoming data. Downstream services like `search` refer to the article store for validation.
- promote the concept of schema even more as a `schema` service. This service would then be able to offer validation to whoever needs it. This implies something we may do anyway, which is versioning of schemas: whenever a new extension like IIIF is introduced, the article schema version number is incremented and the article store and dependants have time to upgrade to deal with it. It is unfeasible to change the schema atomically with the content anyway (think of the hours or even just minutes to regenerate all the affected articles).